### ISSUE TITLE
Can't switch back to FDCL after setting constant constraint value

### ISSUE THREAD

**Issue by grumpyinca** [thegrumpys/odop#480]
Steps to reproduce:

In development environment with current master ...
Start ODOP.
File : Open, select extension spring, Load Initial State.
Choose an FDCL constraint, for example: Force_1 MIN; Click in entry field.
Overwrite the value field with a new value (for example replace 6 with 7); Click Enter button.
Observe expected result in Main Page (Advanced View).
Click on the same FDCL constraint entry field.  
Observe modal offers button to re-establish the FDCL (example: Initial_Tension).
Click that button;
Observe return to main page with no update to the constraint level (example: 7 does not switch to 6)

Is it possible that something got tangled up in the change-over from "pull" to "push" ?

**Comment by 1fifoto** [comment]
Merge into master, closing

**Comment by 1fifoto** [comment]
oops, still not fixed, grrrr

**Comment by 1fifoto** [comment]
Can you see if you can recreate the problem. I think it's fixed. I have git pushed it to odop-staging too.

**Comment by grumpyinca** [comment]
I can't tell if this is a long-standing unrelated problem or something that was recently introduced by your work on this issue.

In development environment,
Click to FIX an extension spring Dependent Variable.
See Traceback: 
```
TypeError: Cannot read property 'undefined' of undefined
(anonymous function)
C:/Users/Mike/git/odop/client/src/store/middleware/dispatcher.js:211
  208 | //        console.log('In dispatcher.RESET_SYMBOL_FLAG.propagate mask=',mask);
  209 |         if (!(mask & FDCL) && (action.payload.mask & FDCL)) { // Remember reducer sets mask before dispatcher (so mask logic's inverted)
  210 |             if (action.payload.minmax === MIN) {
> 211 |                 source = design.model.symbol_table.find(element => element.name === sink.cminchoices[sink.cminchoice]);
  212 |             } else {
  213 |                 source = design.model.symbol_table.find(element => element.name === sink.cmaxchoices[sink.cmaxchoice]);
  214 |             }

```

**Comment by 1fifoto** [comment]
Transfer previous comment's problem to a new issue #498

**Comment by grumpyinca** [comment]
In using the staging system to confirm that this issue is now fixed, I observed a problem that may (or may not) be related to the repair for the original issue.

**Steps to reproduce:**

Reload browser.
File : Open, select extension spring, Load Initial State.
Choose an FDCL constraint Force_1 MIN; Click in entry field.
Overwrite the value field with a new value (for example replace 6 with 7); Click Enter button.
Observe expected result in Main Page (Advanced View).
Click on the same FDCL constraint entry field.
Observe modal offers button to re-establish the FDCL (example: Initial_Tension).
Click that button;
Observe that the value now correctly shows the 6 of Initial_Tension.
File : Open Load InitialState.
Force_1 MIN now shows a value of 3 ?  Something is wrong here !

**Comment by grumpyinca** [comment]
This is a bit of a cosmetic / nit-pick / terminology issue.  I don't know the right answer.  This comment is intended to bring up the question for discussion as opposed to indicate a clear and unambiguous need for change.

After over-riding the FDCL (Initial_Tension = 6) with a constant value of 7, when the cursor is in the Force_1 MIN entry box, a tooltip of "FDCL =7" shows.  Technically, is this a constant value and not a Functionally Determined Constraint Level ?  I understand that since the FDCL of Initial_Tension is still connected and available to be established, the "FDCL" display may be appropriate.  

![image](https://user-images.githubusercontent.com/39974784/106374015-f5029080-6333-11eb-8f82-210df5675e83.png)

**Comment by grumpyinca** [comment]
Work on the initialState problem reported above on January 30 is being carried forward to issue #503.

Other repairs from this branch have been merged into branch #482 and Master.
Closing.
